src/object.d

```diff
@@ -2168,14 +2168,22 @@ const:
      */
     @property string name() nothrow pure @nogc
     {
-        if (true || flags & MIname) // always available for now
+        import core.stdc.string : strlen;
+        static if (1) // always available for now
```

Comment:
And why can't we just remove the `if (1)`?  The assumption that this information is always available is naive at best.

Reply:
Because this PR isn't about changing the behavior of the code, it's clarifying what it does and making it compatible with the compiler's flow analysis.

Changing the behavior is an entirely different thing.

Reply:
Removing an always-true `if` does not change the behavior of the code.

Reply:
Ping @WalterBright: Can you address the reviews so we can merge this ?

Reply:
You're asking me to assume the `true` is not needed, that `flags` is reliably set. That is beyond the scope of this PR.

Reply:
That assumption is *already* there. The code generated before and after removing the `if` will have *exactly the same behavior*. I don't see why you insist on this particular form which just adds dead code.

Reply:
You want to change `if(true || flags)` to `if (flags)`. This is *different*.

Reply:
@WalterBright : My suggestion is:
```D
    @property string name() nothrow pure @nogc
    {
        import core.stdc.string : strlen;
        auto p = cast(immutable char*)addrOf(MIname);
        return p[0 .. strlen(p)];
    }
```

Which is what the code is currently doing.

Reply:
@Geod24 @WalterBright I think that there was a confusion here. @ibuclaw requested that the `true` part of the `if` is removed whereas @Geod24 requested that the `if` is discarded completely. Maybe the resolution is to discard the `if` but add a comment detailing in what circumstances should the `flag` condition be added.

Reply:
> @Geod24 @WalterBright I think that there was a confusion here. @ibuclaw requested that the `true` part of the `if` is removed whereas @Geod24 requested that the `if` is discarded completely. Maybe the resolution is to discard the `if` but add a comment detailing in what circumstances should the `flag` condition be added.

Or raise a new PR making the if explicit, because it allows this to be a PAYG feature, the same as all other getter methods for minfo, and avoids segfault when the compiler/user decides to omit giving a module a name at runtime.